cmd/controller/main_test.go

```diff
@@ -29,9 +32,20 @@ func hasAction(fake *fake.Clientset, verb, resource string) bool {
 	return findAction(fake, verb, resource) != nil
 }
 
+// generateNameReactor implements the logic required for the GenerateName field to work when using
+// the fake client. Add it with client.PrependReactor to your fake client.
+func generateNameReactor(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+	s := action.(ktesting.CreateAction).GetObject().(*v1.Secret)
+	if s.Name == "" && s.GenerateName != "" {
+		s.Name = fmt.Sprintf("%s-%d", s.GenerateName, mathrand.Intn(10000))
```

Comment:
This is probably good enough but may cause super weird failures if the same int is generated 2+ times.

Maybe it would be better to use a per-test generator? Would a simple counter then be sufficient to provide uniqueness?

But if you stick with unique, does it need to be `Seed`ed?

Reply:
what if we just use the same random string function that k8s uses to generate ranm pod names? (yes there is still the theoretical chance of conflict, but if that was the only source of flakiness I think I can live with that)

Reply:
Works for me. I really just wanted to point out the potential issue in case it was likely to cause issues.

Reply:
yeah indeed 1000 was too small a number